include/pybind11/eigen/tensor.h

```diff
@@ -328,12 +335,8 @@ struct type_caster<Type, typename eigen_tensor_helper<Type>::ValidType> {
     }
 };
 
-template <class T>
-struct is_const_pointer {};
-template <class T>
-struct is_const_pointer<T *> : std::false_type {};
-template <class T>
-struct is_const_pointer<const T *> : std::true_type {};
+template<typename T>
+using is_pointer_to_const = std::is_const<typename std::remove_pointer<T>::type>;
```

Comment:
Actually, thinking again this isn't the best name as this will also evaluate to true if it's not actually a pointer but it is const. You could make it a constexpr function with a static_assert checking if it's a pointer or have some additional logic which will satisfy the need_writable otherwise

Reply:
I'll just manually inline this. It will always be called on a pointer so it should be OK.